fix sflow ipv4 and ipv6 data layer decoding

sharath-vemula opened the pull request:
according to https://sflow.org/sflow_version_5.txt, sflow sampled_ipv4 and sampled_ipv6 also has
SFlowBaseFlowRecord which is also evident from the solarwinds sflow pcap. 
Corrected the decoding logic to incorporate SFlowBaseFlowRecord

ericyitaowu commented on layers/sflow.go:
```diff
@@ -1832,6 +1832,59 @@ func decodeExtendedUserFlow(data *[]byte) (SFlowExtendedUserFlow, error) {
+//  /                                               /
+//  +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
+
+type SFlowIpv4DataRecord struct {
```
Can we modify the existing SFlowIpv4Record directly? Is there any concern not to touch the original code?

sharath-vemula replied:
Ack

ericyitaowu commented on layers/sflow_test.go:
```diff
@@ -13,6 +13,59 @@ import (
 )
 
+// Test packet collected from PCAP from SolarWinds tool. See the test below for contents
+var SFlowTestPacketWithIPv4Data = []byte{
```
Can you trim the test data to include IPv4 sample only, no counter sample? It should be easy to do so. You just need to modify a few bytes.

sharath-vemula replied:
Ack

sharath-vemula committed:
fix sflow ipv4 and ipv6 data layer decoding

according to https://sflow.org/sflow_version_5.txt, sflow sampled_ipv4 and sampled_ipv6 also has
SFlowBaseFlowRecord which is also evident from the solarwinds sflow pcap.
Corrected the decoding logic to incorporate SFlowBaseFlowRecord

ericyitaowu commented:
looks good
